### ISSUE TITLE
Cannot find stdcc++ and gcc_s

### ISSUE THREAD

**Issue by junland** [mikpe/buildcross#1]
When building `x86_64-unknown-linux-gnu`, process fails at function `build_target_libc` with this error:

```
/home/ubuntu/buildcross/crossdir/lib/gcc/x86_64-unknown-linux-gnu/10.1.0/../../../../x86_64-unknown-linux-gnu/bin/ld: cannot find -lstdc++
/home/ubuntu/buildcross/crossdir/lib/gcc/x86_64-unknown-linux-gnu/10.1.0/../../../../x86_64-unknown-linux-gnu/bin/ld: cannot find -lgcc_s
collect2: error: ld returned 1 exit status
```

_Note: Crossdir is set to `/home/ubuntu/buildcross`_

This was built on a fresh installation of Ubuntu 20.04 LTS with the below packages:

```
rsync build-essential pkg-config libarchive-tools m4 gawk bc bison flex texinfo python3 perl libtool autoconf automake autopoint autoconf-archive mtools liblzma-dev libelf-dev libssl-dev zlib1g-dev zlib1g xz-utils lzip
```

Suspecting its because either the `make` or `install` targets for that function need to be something like this?

```
make all-gcc all-target-libgcc
make install-gcc install-target-libgcc
```

Hope this helps. Love this script by the way.

**Comment by mikpe** [comment]
I can't reproduce this error with buildcross v3.7.0 on Fedora 31, invoked as ~/proj/buildcross.git/buildcross -j1 x86_64 from /tmp.
The error indicates it's trying to link something with the shared libgcc and libstdc++, which shouldn't happen since I deliberately configure both host and cross gcc:s with --disable-shared.
At this point I suspect something in binutils, gcc, or glibc detects that you're on Ubuntu and silently changes something, but I cannot tell what.

**Comment by junland** [comment]
@mikpe Interesting! I'll have to take a look at that and try in Fedora and see what that does.

**Comment by mikpe** [comment]
I was able to reproduce the issue after booting the Ubuntu 20.04 image off a usb disk, and then installing most of the packages you mentioned (apt complained about some of them not being available).  It appears that Ubuntu in this state has g++ but not the necessary library bits to support -m32.  Glibc detects g++ and decides to build its `support` library as C++, which it then tries to link against the shared libstdc++ and libgcc_s, but those components aren't present.

Obviously my non-vanilla Fedora environment includes all those things so the build succeeds.

Either I'll have to figure out how to force-disable glibc building `support` with g++, or I'll have to build the host gcc with shared libraries.  I really don't like the 2nd option.

I also hade problems with binutils-2.34 as it required makeinfo which my fresh Ubuntu 20.04 didn't have and could't find; I ended up downgrading to binutils-2.33.1 in the buildcross script.

I'll have to think about this one for a while.

**Comment by junland** [comment]
Interesting, even deleting the below lines of code to build 32-bits on line `535` doesn't seem to get past the issue.

```
case "${TARGET_ARCH}" in
	"mips64")
	    build_glibc "mips-unknown-linux-gnu" "CC=mips64-unknown-linux-gnu-gcc -mabi=32"
	    build_glibc "mips-unknown-linux-gnu" "CC=mips64-unknown-linux-gnu-gcc -mabi=n32"
	    TARGET_CC="CC=mips64-unknown-linux-gnu-gcc -mabi=64"
	    ;;
	"ppc64")
	    build_glibc "ppc-unknown-linux-gnu" "CC=ppc64-unknown-linux-gnu-gcc -m32"
	    ;;
	"s390x")
	    build_glibc "s390-unknown-linux-gnu" "CC=s390x-unknown-linux-gnu-gcc -m31"
	    ;;
	"sparc64")
	    build_glibc "sparcv9-unknown-linux-gnu" "CC=sparc64-unknown-linux-gnu-gcc -m32 -mcpu=ultrasparc"
	    ;;
	"x86_64")
	    build_glibc "i686-unknown-linux-gnu" "CC=x86_64-unknown-linux-gnu-gcc -m32"
	    build_glibc "x86_64-unknown-linux-gnu" "CC=x86_64-unknown-linux-gnu-gcc -mx32"
	    ;;
    esac
```

Though I'm able to use `crosstool-ng` to build a plain x86_64 toolchain just fine.

**Comment by mikpe** [comment]
Should be fixed by just pushed v3.8.0.  Key points: downgrade binutils to 2.33.1 to avoid makeinfo dependency, and configure target glibc with CXX=/bin/false to force-disable its use of C++ and related shared libraries.  Tested on Ubuntu 20.04 fresh install with apt install rsync build-essential pkg-config m4 gawk bc bison flex python3 perl libtool autoconf automake autopoint mtools liblzma-dev libelf-dev libssl-dev zlib1g-dev zlib1g xz-utils (the other packages mentioned couldn't be found).

**Comment by junland** [comment]
Awesome! Thanks a lot for taking a look at this!
